### PR TITLE
Add slidingwindow to Conda

### PR DESCRIPTION
**Pull request by henrykironde**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by henrykironde** [comment]
@conda-forge/help-python we are waiting for the maintainer to comment for step 8

**Comment by bw4sz** [comment]
@henrykironde the conda mantainer implied that all numpy based packages need that line 

https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

```
host:
  - numpy
run:
  - {{ pin_compatible('numpy') }}
```

**Comment by isuruf** [comment]
Please revert the last line. Docs clearly say that the line is needed if the package links to numpy and also gives some instructions on how to do this.

**Comment by isuruf** [comment]
> we are waiting for the maintainer to comment for step 8

The maintainer should be you. Not the author of the package.

**Comment by marcelotrevisani** [comment]
> @henrykironde the conda mantainer implied that all numpy based packages need that line
> 
> https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy
> 
> ```
> host:
>   - numpy
> run:
>   - {{ pin_compatible('numpy') }}
> ```

I think this package is not compiling against ``numpy``, at least it does not seem to be doing it. It should be just ``numpy``

File: recipes/slidingwindow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    - "{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv "
+
+requirements:
+  host:
+    - python
+    - pip
+
+  run:
+    - python
+    - {{ pin_compatible('numpy') }}
```
Comment (marcelotrevisani) [inline recipes/slidingwindow/meta.yaml:26]:
```suggestion
    - numpy
```

File: recipes/slidingwindow/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script:
+    - export SETUPTOOLS_SCM_PRETEND_VERSION={{ version }}    # [unix]
```
Comment (marcelotrevisani) [inline recipes/slidingwindow/meta.yaml:16]: do you really need this line?
I didn't see any reference to ``setuptools_scm`` in this project, but I might be missing something
```suggestion
```

File: recipes/slidingwindow/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
```
Comment (marcelotrevisani) [inline recipes/slidingwindow/meta.yaml:14]:
```suggestion
  number: 0
  noarch: python
```

Removing the line which is exporting the that variable you can add this recipe as noarch

File: recipes/slidingwindow/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    that utilise a sliding window to break the processing up into a
+    series of smaller chunks. In addition, a set of optional transformations
+    can be specified to be applied to each window.
+
+  doc_url: https://github.com/adamrehn/slidingwindow/blob/master/README.md
+  dev_url: https://github.com/adamrehn/slidingwindow
+
+extra:
+  recipe-maintainers:
+    - adamrehn
```
Comment (marcelotrevisani) [inline recipes/slidingwindow/meta.yaml:51]: As @isuruf explained, this line should be at least you and not the developer of this project
```suggestion
    - adamrehn
    - henrykironde
```

**Comment by henrykironde** [comment]
I think I have covered all

**Comment by bw4sz** [comment]
@marcelotrevisani or @isuruf thanks for your help. It looks like this ready.

File: recipes/slidingwindow/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
+{% set name = "slidingwindow" %}
+{% set version = "0.0.14" %}
+{% set sha256 = "4c52adf24e2dd6786b9c1c49144083254d9bd9de655b4f2f17bd78cc41a01e49" %}
```
Comment (chrisburr) [inline recipes/slidingwindow/meta.yaml:3]:
```suggestion
```

File: recipes/slidingwindow/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: {{ sha256 }}
```
Comment (chrisburr) [inline recipes/slidingwindow/meta.yaml:11]: No need to use a variable as it's only used in one place:
```suggestion
  sha256: 4c52adf24e2dd6786b9c1c49144083254d9bd9de655b4f2f17bd78cc41a01e49
```

**Pull request merged**